src/executor/exec_pass.h

```diff
@@ -189,6 +189,13 @@ bool DefaultStorageType(const nnvm::NodeAttrs& attrs,
                         std::vector<int> *iattr,
                         std::vector<int> *oattr);
 
+/*! \brief The default type inference function, which assigns all undefined
+ *         types to the same type of one of the inputs or outputs.
+ */
+bool SameType(const nnvm::NodeAttrs& attrs,
```

Comment:
where is the definition of this function?

Reply:
I haven’t found the definition of SameType. It seems that this function is just defined and implemented in infer_graph_attr_pass.cc.

A same kind of function, DefaultStorageType, is defined in this header and implemented in infer_graph_attr_pass.cc.

Reply:
I tried global search SameType in github. It only appears twice in infer_graph_attr_pass.cc https://github.com/apache/incubator-mxnet/search?utf8=✓&q=SameType&type=

Reply:
Can we move the definition to some file in `src/common` for better code management? It's kind of confusing to declare it here and define it in a file not related to the caller.

Reply:
Sure. Will update it tomorrow.